libctru/source/services/soc/soc_getaddrinfo.c

```diff
@@ -52,7 +52,14 @@ static struct addrinfo * buffer2addrinfo(addrinfo_3ds_t * entry)
 		ai->ai_addrlen  = entry->ai_addrlen;
 
 		memcpy(ai->ai_canonname, entry->ai_canonname, ai_canonname_len);
+
+		// make sure we account for sin_zero, which the hos-native sockaddr_in doesn't have
+		memset(ai->ai_addr, 0, sizeof(struct sockaddr_storage));
 		memcpy(ai->ai_addr, &entry->ai_addr, ai->ai_addrlen);
+
+		if(ai->ai_family == AF_INET)
+			ai->ai_addrlen += 8;
```

Comment:
Is there any reason why this can't just be `ai->ai_addrlen = sizeof(struct sockaddr_in)`?

Reply:
I don't see any reason it can't be that.

I just did it the same way #550 did it in several places. Should I then change those to `= sizeof(struct sockaddr_in)` for consistency, or only update it in here?

Reply:
We should probably change the other functions as well in order to make the code clearer & not use hardcoded constants. But for the sake of this PR, I think it's OK to only limit the change to this function.